### ISSUE TITLE
Error reading temperature value ['temp_c']

### ISSUE THREAD

**Issue by rudibarani** [xfjx/homebridge-wunderground#3]
Hi,
i am getting an error message indicating that the plugin might have a problem with temperature values in degrees Celsius:

2016-11-02 21:03:10] [Wetter Berlin] Successfully fetched weather data from wunderground.com
/usr/local/lib/node_modules/homebridge-wunderground/index.js:36
    		that.temperature = response['current_observation']['temp_c'];
    		                                                                                ^

TypeError: Cannot read property 'temp_c' of undefined
    at Wunderground.<anonymous> (/usr/local/lib/node_modules/homebridge-wunderground/index.js:36:57)
    at Request._callback (/usr/local/lib/node_modules/homebridge-wunderground/node_modules/wundergroundnode/lib/wundergroundnode.js:137:26)
    at Request.self.callback (/usr/local/lib/node_modules/homebridge-wunderground/node_modules/request/request.js:186:22)
    at emitTwo (events.js:106:13)
    at Request.emit (events.js:191:7)
    at Request.<anonymous> (/usr/local/lib/node_modules/homebridge-wunderground/node_modules/request/request.js:1060:10)
    at emitOne (events.js:96:13)
    at Request.emit (events.js:188:7)
    at IncomingMessage.<anonymous> (/usr/local/lib/node_modules/homebridge-wunderground/node_modules/request/request.js:980:12)
    at IncomingMessage.g (events.js:291:16)
    at emitNone (events.js:91:20)
    at IncomingMessage.emit (events.js:185:7)
    at endReadableNT (_stream_readable.js:974:12)
    at _combinedTickCallback (internal/process/next_tick.js:74:11)
    at process._tickCallback (internal/process/next_tick.js:98:9)

**Comment by elysiandc** [comment]
+1 for this - exactly the same error here. My config entry is
`    "accessories": [
        {
            "accessory": "WUNDERGROUND",
            "name": "Weather Underground",
            "device": "WUBelgraveSouth",
            "key": "6c8f4bd1971c5e7b",
            "city": "Australia/Belgrave South"
         }
    ]
`

**Comment by elysiandc** [comment]
OK - I found the problem with mine. Underground was not returning the expected JSON array with for the value I had set for 'city'. This was causing this plugin to die and crash HomeBridge, causing it to get stuck in a loop of restarting every minute or so.

First issue was having a space in the town name and '%20' was no good. Seems the standard is an '_' to replace spaces.

Next was that Underground wasn't able to return details for the location that I wanted. I specifically wanted results for my personal weather station. To achieve that the correct format for the 'city' parameter was 'pws:STATION_ID', where STATION_ID is the unique identifier given to each personal weather station.

@rudibarani I think one of these is likely to be the problem you are having.

@xfjx  I think some data validation and error handling so that the scrip exits gracefully could be useful. I'd contribute but have never used Github before.

Hope that helps.
